lib/src/merged_tree.rs

```diff
@@ -321,6 +321,41 @@ impl MergedTree {
     ) -> TreeDiffIterator<'matcher> {
         TreeDiffIterator::new(self.clone(), other.clone(), matcher)
     }
+
+    /// Merges this tree with `other`, using `base` as base.
+    pub fn merge(
+        &self,
+        base: &MergedTree,
+        other: &MergedTree,
+    ) -> Result<MergedTree, TreeMergeError> {
+        if let (MergedTree::Legacy(this), MergedTree::Legacy(base), MergedTree::Legacy(other)) =
+            (self, base, other)
+        {
+            let merged_tree = tree::merge_trees(this, base, other)?;
+            Ok(MergedTree::legacy(merged_tree))
+        } else {
+            // Convert
+            let to_merge = |tree: &MergedTree| -> Merge<Tree> {
+                match tree {
+                    MergedTree::Legacy(tree) => {
+                        let MergedTree::Merge(tree) = MergedTree::from_legacy_tree(tree.clone())
+                        else {
+                            unreachable!();
+                        };
+                        tree
+                    }
+                    MergedTree::Merge(conflict) => conflict.clone(),
+                }
+            };
+            let nested = Merge::new(vec![to_merge(base)], vec![to_merge(self), to_merge(other)]);
+            let tree = MergedTree::Merge(nested.flatten().simplify()).resolve()?;
```

Comment:
nit: I slightly prefer calling `merge_trees()` directly than going back to `MergedTree` abstraction.

Reply:
`merge_trees()` produces `TreeValue::Conflict` values in case of conflicts, and we don't want to have such values in the result. Or am I just not following?

Btw, I updated the comment above a little - not to clarify this point, but because I think it was supposed to say more than just "Convert".

Reply:
I mean `merged_tree::merge_trees()` in place of `MergedTree::Merge(..).resolve()` (because this function processes `Merge<Tree>` internally.)

Reply:
Ah, I had forgotten about `merged_tree::merge_trees()`. Yes, that makes more sense, thanks. Will fix.